optuna/samplers/nsgaii/_sampler.py

```diff
@@ -410,29 +410,46 @@ def after_trial(
 
 
 def _calc_crowding_distance(population: List[FrozenTrial]) -> DefaultDict[int, float]:
+
     manhattan_distances: DefaultDict[int, float] = defaultdict(float)
+    if len(population) == 0:
+        return manhattan_distances
+
     for i in range(len(population[0].values)):
         population.sort(key=lambda x: cast(float, x.values[i]))
 
-        v_min = population[0].values[i]
-        v_max = population[-1].values[i]
-        assert v_min is not None
-        assert v_max is not None
-
-        width = v_max - v_min
-        if width == 0:
+        # If population have the same values[i], ignore that value.
+        if population[0].values[i] == population[-1].values[i]:
             continue
 
-        manhattan_distances[population[0].number] = float("inf")
-        manhattan_distances[population[-1].number] = float("inf")
+        vs = (
+            [-float("inf")]
+            + [cast(float, population[j].values[i]) for j in range(len(population))]
+            + [float("inf")]
+        )
+
+        # Smallest finite value.
+        v_min = next(x for x in vs if x != -float("inf"))
```

Comment:
I'm not sure, but could you tell me what happens when objective values of all individuals are `-inf`?

Reply:
In that case, it will not reach this line because of L422.

Reply:
That makes sense. Thanks.